src/main.rs

```diff
@@ -1,15 +1,17 @@
-use beacon_api_client::{Client, StateId};
+pub mod api_client;
 use url::Url;
+pub mod cli;
+pub mod error;
+pub mod serde;
+pub mod types;
+use clap::Parser;
```

Comment:
most of this is saying "if I am processing rust code from `src/main.rs`, there are modules `cli`, `error`, ..."

which is not what we want

`src/main.rs` files produce binaries by default which should be structured as programs that consume libraries;

this means that we should only need:

```suggestion
use url::Url;
use clap::Parser;
use beacon_api_client::{Client, CliConfig, run_cli};
```